File: python/006.py

def solve1(x):
    return sum(map(lambda a: pow(a, 2), range(x + 1)))


def solve2(x):
    return pow(sum(range(x + 1)), 2)


if __name__ == '__main__':
    assert solve1(10) == 385
    assert solve2(10) == 3025
    assert solve2(10) - solve1(10) == 2640

    assert solve2(100) - solve1(100) == 25164150

File: ruby/006.rb
def solve1(x)
    (0..x).reduce(0){|sum, a| sum+=a**2}
end

def solve2(x)
    (0..x).reduce(:+) ** 2
end

def solve(x)
    solve2(x) - solve1(x)
end

require 'test/unit'
class TestSolve < Test::Unit::TestCase
    def test_solve1
        assert solve1(10) == 385
    end

    def test_solve2
        assert solve2(10) == 3025
    end

    def test_solve
        assert solve(10) == 2640
        assert solve(100) == 25164150
    end
end
